apps/root/src/state/balances/actions.ts

```diff
@@ -1,121 +1,130 @@
   const tokens = Object.values(tokenList);
 
-  let balances: Record<number, Record<string, BigNumber>> = {};
-  try {
-    balances = await sdkService.getMultipleBalances(tokens, walletAddress);
-  } catch (e) {
-    console.error(e);
-  }
+  const balances = await sdkService.getMultipleBalances(tokens, walletAddress);
 
-  const tokenAccountBalances: TokenBalancesAndPrices = {};
-  Object.entries(balances[chainId]).forEach(([tokenAddress, balance]) => {
+  const tokenBalances = Object.entries(balances[chainId]).reduce((acc, [tokenAddress, balance]) => {
     if (balance.gt(0)) {
-      tokenAccountBalances[tokenAddress] = {
-        token: tokenList[tokenAddress],
-        balances: {
-          [walletAddress]: balance,
+      return {
+        ...acc,
+        [tokenAddress]: {
+          token: tokenList[tokenAddress],
+          balances: {
+            [walletAddress]: balance,
+          },
         },
       };
     }
-  });
+    return acc;
+  }, {} as TokenBalancesAndPrices);
 
-  return { chainId, tokenBalances: tokenAccountBalances };
+  return { chainId, tokenBalances };
 });
 
 export const fetchPricesForChain = createAsyncThunk<
   { chainId: number; prices: CurrentPriceForChainResponse },
   { chainId: number },
   { extra: ExtraArgument }
->('prices/fetchPricesForChain', async ({ chainId }, { extra: { web3Service }, getState, dispatch }) => {
+>('prices/fetchPricesForChain', async ({ chainId }, { extra: { web3Service }, getState }) => {
   const sdkService = web3Service.getSdkService();
   const state = getState() as RootState;
   const storedTokenAddresses = Object.values(state.balances[chainId].balancesAndPrices || {}).map(
     (tokenBalance) => tokenBalance.token.address
   );
   let priceResponse: CurrentPriceForChainResponse = {};
   if (!!storedTokenAddresses.length) {
-    try {
-      dispatch(setLoadingChainPrices({ chainId, isLoading: true }));
-      priceResponse = await sdkService.sdk.priceService.getCurrentPricesForChain({
-        chainId,
-        addresses: storedTokenAddresses,
-      });
-    } catch (e) {
-      console.error(e);
-    }
-    dispatch(setLoadingChainPrices({ chainId, isLoading: false }));
+    priceResponse = await sdkService.sdk.priceService.getCurrentPricesForChain({
+      chainId,
+      addresses: storedTokenAddresses,
+    });
   }
 
   return { chainId, prices: priceResponse };
 });
 
 export const fetchPricesForAllChains = createAsyncThunk<void, void, { extra: ExtraArgument }>(
   'balances/fetchInitialBalances',
-  async (nothing, { getState, dispatch }) => {
+  (_, { getState, dispatch }) => {
     const state = getState() as RootState;
     const chainsWithBalance = Object.keys(state.balances);
 
-    const pricePromises = chainsWithBalance.map(async (chainId) =>
+    const pricePromises = chainsWithBalance.map((chainId) =>
       dispatch(fetchPricesForChain({ chainId: Number(chainId) }))
     );
-    await Promise.all(pricePromises);
+    void Promise.all(pricePromises);
```

Comment:
If we want the pending to be correctly calculated, we should await this promises